File: app/src/main/java/com/gmail/fattazzo/formula1world/fragments/current/races/RacesListAdapter.java
package com.gmail.fattazzo.formula1world.fragments.current.races;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import com.gmail.fattazzo.formula1world.domain.F1Race;

import org.androidannotations.annotations.AfterInject;
import org.androidannotations.annotations.EBean;
import org.androidannotations.annotations.RootContext;
import org.apache.commons.collections4.CollectionUtils;

import java.util.ArrayList;
import java.util.List;

@EBean
class RacesListAdapter extends BaseAdapter {

    @RootContext
    Context context;

    private List<F1Race> races;

    public void setRaces(List<F1Race> races) {
        CollectionUtils.addAll(this.races,CollectionUtils.emptyIfNull(races));
    }

    @AfterInject
    void initAdapter() {
        races = new ArrayList<>();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        RaceItemView raceItemView;
        if (convertView == null) {
            raceItemView = RaceItemView_.build(context);
        } else {
            raceItemView = (RaceItemView) convertView;
        }

        raceItemView.bind(getItem(position));

        return raceItemView;
    }

    @Override
    public int getCount() {
        return races.size();
    }

    @Override
    public F1Race getItem(int position) {
        return races.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    void clearItems() {
        races.clear();
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/fragments/current/races/RacesListAdapter.kt
/*
 * Project: total-gp-world
 * File: RacesListAdapter.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.fragments.current.races

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.gmail.fattazzo.formula1world.domain.F1Race
import org.androidannotations.annotations.AfterInject
import org.androidannotations.annotations.EBean
import org.androidannotations.annotations.RootContext
import org.apache.commons.collections4.CollectionUtils
import java.util.*

@EBean
internal open class RacesListAdapter : BaseAdapter() {

    @RootContext
    lateinit var context: Context

    private var races: MutableList<F1Race>? = null

    fun setRaces(races: List<F1Race>) {
        CollectionUtils.addAll(this.races, CollectionUtils.emptyIfNull(races))
    }

    @AfterInject
    fun initAdapter() {
        races = ArrayList()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val raceItemView: RaceItemView = if (convertView == null) {
            RaceItemView_.build(context)
        } else {
            convertView as RaceItemView
        }

        raceItemView.bind(getItem(position))

        return raceItemView
    }

    override fun getCount(): Int {
        return races!!.size
    }

    override fun getItem(position: Int): F1Race {
        return races!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    fun clearItems() {
        races!!.clear()
    }
}